### ISSUE TITLE
Windows: Cancellation and Logs Streaming Behavior

### ISSUE THREAD

**Issue by amitsaha** [buildkite/agent#797]
What is the expected behavior of `stdout` streaming when a job is cancelled? This is what I am seeing on a Windows agent:

- Build cancelled
- Agent keeps streaming logs (the build did produce a huge chunk of logs when it was running)
- Another job submitted and assigned to the agent doesn't get picked up

Manually stopping/starting the agent "fixes" it.

**Comment by lox** [comment]
Canceling should terminate the bootstrap, which should stop new output. The agent will then batch up the previous output into chunks and upload them, which will continue to run after cancellation. A new job won't be picked until the chunks are uploaded. Are you seeing something else?

**Comment by amitsaha** [comment]
@lox thanks for the clarification. That is indeed what I am seeing. Closing.

**Comment by amitsaha** [comment]
I am going to dig this issue up so that I understand this better - specifically on Windows. AFAIK, this is the current process tree for the agent:

```
- agent 
   - <new job>
   - bootstrap process
     - <user-defined scripts>
       - <any other processes>
  - <job end>
```

What I was seeing prior to https://github.com/buildkite/agent/pull/795 was that cancelling a job sometimes kept the agent occupied for minutes and hence not picking up a new job. However, post #795 - this is not an issue any more. Cancellation finishes in seconds and new job gets picked up without any issues at all.

I am wondering if there is anything else at play here that the above change has "fixed"?

**Comment by lox** [comment]
Closed via #879 and #860.
